projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -5437,10 +5453,6 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
     get filteredSortedData(): any[] {
         return this._filteredSortedData;
     }
-    set filteredSortedData(value: any[]) {
```

Comment:
You can here use this.refreshSearch(true, false);, without adding the new method setFilteredSortedData

Reply:
The aim is to remove this public setter. Since we cant hide the setter and leave the getter only, I removed the native setter and use this hidden method `setFilteredSortedData`